discovery-provider/src/utils/multi_provider.py

```diff
@@ -0,0 +1,18 @@
+import random
+from web3.providers import HTTPProvider, BaseProvider
+
+
+class MultiProvider(BaseProvider):
+    def __init__(self, providers):
+        self.providers = [HTTPProvider(provider) for provider in providers.split(",")]
```

Comment:
might be worth overriding the __str__ as well here
https://github.com/ethereum/web3.py/blob/e47ff679398c05b72fdb60d07719dfc9ab04c245/web3/providers/rpc.py#L72